api/api.go

```diff
@@ -112,13 +112,14 @@ type DatasetAPI struct {
 	auditor                Auditor
 	enablePrivateEndpoints bool
 	enableDetachDataset    bool
-	datasetAuth            AuthHandler
+	datasetPermissions     AuthHandler
+	permissions            AuthHandler
```

Comment:
I get permissions vs datasetPermissions right now but not sure how well it'd hold up in the long term (can't think of anything obviously better though, instancePermissions maybe?).

Reply:
It's not strictly instance permissions though (this is the dilemma I faced). The problem is to authorise access to a dataset you need additional information compared to instance authorisation which is why there 2 versions of it. I felt it was much too complicated to have a single auth func that did all the things. But like you I've been unable to come up with a better name.